### PR TITLE
added L0 for file Encoding

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

File: Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+try {
+    var fileEncodeType = fileEncoding.detectFileEncoding('utf16betxt', new Buffer([254, 255, 0 ,0]));
+}
+catch(exception) {
+    console.log('UTF-16BE with BOM validated');
+}
+
+var fileEncodeType = fileEncoding.detectFileEncoding('utf8.txt', new Buffer([255, 254, 10, 10]));
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts]: utf-18le

File: Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+try {
+    var fileEncodeType = fileEncoding.detectFileEncoding('utf32le.txt', new Buffer([255, 0, 0, 0]));
+}
+catch(exception) {
+    console.log('UTF-32LE without BOM validated');
+}
+
+try {
+    var fileEncodeType = fileEncoding.detectFileEncoding('utf32le.txt', new Buffer([0, 0, 0, 255]));
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts]: utf32be.txt

File: Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+try {
+    var fileEncodeType = fileEncoding.detectFileEncoding('utf16be.txt', new Buffer([0, 10, 0 ,20]));
+}
+catch(exception) {
+    console.log('UTF-16BE without BOM validated');
+}
+
+var fileEncodeType = fileEncoding.detectFileEncoding('utf8.txt', new Buffer([20, 0, 10, 0]));
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts]: utf16le.txt

File: Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts (hunk 1 above)
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts]: done

File: Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts (hunk 3 above)
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts]: done

File: Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts (hunk 2 above)
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/Tests/L0ValidateFileEncoding.ts]: done

**Pull request merged**